packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -722,7 +736,14 @@ class SliverReorderableListState extends State<SliverReorderableList> with Ticke
       if (item.index == _dragIndex! || !item.mounted)
         continue;
 
-      final Rect geometry = item.targetGeometry();
+      Rect geometry = item.targetGeometry();
+      if (!_dragStartTransitionComplete && _dragIndex! <= item.index) {
+        // Transition is not complete, so each item after the dragged item is still
+        // in its normal location and not moved up for the zero sized box that will
+        // replace the dragged item.
+        final Offset transitionOffset = _extentOffset(_reverse ? -gapExtent : gapExtent, _scrollDirection);
+        geometry = (geometry.topLeft - transitionOffset) & geometry.size;
+      }
```

Comment:
Just set _dragStartTransitionComplete to true here, instead of using a post-frame callback?

Reply:
I thought about that, but the issue is that we don't know when this first update will occur. It could happen after the first frame, in which case we don't want to recompute the offset as it is already correct. Using this flag is just a way of making sure this is only recomputed if we are still before the first frame of the drag which is when we need to compensate for the dragged item that is still in the list.

Reply:
Another alternative I played with was to reset the flag from the `build` method of the dragged widget (when it was actually being replaced with the zero sized box). That seemed messier because it meant the child item widget was directly manipulating the state in the list (or I would need to expose a method). The post-frame callback seemed the most self contained.

Reply:
I see, thanks for the detailed explanation!

Reply:
Thanks for the review.